lib/dockerregistry/inventory.go

```diff
@@ -2547,22 +2570,41 @@ func (m Manifest) Contains(gcrPayload GCRPubSubPayload) bool {
 			if !strings.Contains(gcrPayload.Digest, (string)(image.ImageName)) {
 				continue
 			}
+			r |= FlagPathMatched
 			for digest, tags := range image.Dmap {
 				fqin := ToFQIN(rc.Name, image.ImageName, digest)
-				if fqinOnly {
+				if len(gcrPayload.Tag) == 0 {
+					// The gcrPayload.Digest field actually holds the FQIN.
 					if gcrPayload.Digest == fqin {
-						return true
+						r |= FlagDigestMatched
```

Comment:
the same comment about bitwise operator as above, let's discuss if maybe we could provide some information about how it will behave for less experienced contributors?